### DISCUSSION TITLE
Dúvidas de modelação de domínio

### DISCUSSION THREAD

**Discussion by matheusvarela** [balta-io/forum#231] [Tira Dúvidas] [answered]
Boa noite Balta

Sempre baseie no desenvolvimento dirigido por dados, agora estou tentando (kkk) fazer por domínio, estou com algumas dúvidas

Quando a entidade tem propriedades privadas e usa construtores para "setar" os dados, é possível "carregar"  dados nas as entidades sem passar pelo construtor? como fica o entity framework/dapper, ele consegue "setar" dados nas propriedades mesmo sendo private? 

Para realizar alteração preciso antes carregar a entidade com dados vindo do repositório ou posso dar new na entidade com dados 
 vindo de uma viewmodel que foi previamente carregada com dados de uma entidade? Se a primeira opção for mais indicada teria que criar métodos "sets" para propriedades?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @matheusvarela, como vai?

_Quando a entidade tem propriedades privadas e usa construtores para "setar" os dados, é possível "carregar" dados nas as entidades sem passar pelo construtor? como fica o entity framework/dapper, ele consegue "setar" dados nas propriedades mesmo sendo private?_
- Tem sim! Como eu comento nas aulas de EF, ele cria um proxy da classe e assim consegue não passar pelo construtor.

_Para realizar alteração preciso antes carregar a entidade com dados vindo do repositório ou posso dar new na entidade com dados
vindo de uma viewmodel que foi previamente carregada com dados de uma entidade?_ 
- Precisa carregar antes... é o processo conhecido como re-hidratação.

_Se a primeira opção for mais indicada teria que criar métodos "sets" para propriedades?_
- Sim, se a propridade pode ser modificada, precisa do set... porém ele pode ser privado, como mostro nos cursos Modelando Domínios Ricos, Refatorando para Testes de Unidade e Aplicando OOP em cenários reais.
